### ISSUE TITLE
[firebase_analytics]: Error: The pod required by the plugin requires a higher minimum macOS deployment version

### ISSUE THREAD

**Issue by charneykaye** [firebase/flutterfire#16754]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Which plugins are affected?

Analytics

### Which platforms are affected?

macOS

### Description

I cannot build my Flutter project for Macos after installing the `firebase_analytics` package.

Fails with 

> Error: The pod "Firebase/Analytics" required by the plugin "firebase_analytics" requires a higher minimum macOS deployment version than the plugin's reported minimum version.

### Reproducing the issue

I reproduced the issue from a completely blank project in only 2 steps. Uploaded the project [here](https://github.com/charneykaye/fba_demo)

**1. Flutter create project**

Create a new project `fba_demo`

Flutter 3.24.5 • channel stable • https://github.com/flutter/flutter.git
Framework • revision dec2ee5c1f (11 days ago) • 2024-11-13 11:13:06 -0800
Engine • revision a18df97ca5
Tools • Dart 3.5.4 • DevTools 2.37.3

**2. Flutter add firebase_analytics and firebase_core**

Add the following to pubspec.yaml

```
dependencies:
  firebase_analytics: ^11.3.5
  firebase_core: ^3.8.0
```

**3. Flutter run macos**

### Firebase Core version

3.8.0

### Flutter Version

3.24.5

### Relevant Log Output

```shell
$ flutter run -d macOS

Launching lib/main.dart on macOS in debug mode...
Running pod install...
CocoaPods' output:
↳
Preparing

    Analyzing dependencies

    Inspecting targets to integrate
      Using `ARCHS` setting to build architectures of target `Pods-Runner`: (``)
      Using `ARCHS` setting to build architectures of target `Pods-RunnerTests`: (``)

    Fetching external sources
    -> Fetching podspec for `FlutterMacOS` from `Flutter/ephemeral`
    -> Fetching podspec for `firebase_analytics` from `Flutter/ephemeral/.symlinks/plugins/firebase_analytics/macos`
    firebase_analytics: Using Firebase SDK version '11.4.0' defined in 'firebase_core'
    -> Fetching podspec for `firebase_core` from `Flutter/ephemeral/.symlinks/plugins/firebase_core/macos`
    firebase_core: Using Firebase SDK version '11.4.0' defined in 'firebase_core'

    Resolving dependencies of `Podfile`
      CDN: trunk Relative path: CocoaPods-version.yml exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_0_3_5.txt exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/0/3/5/Firebase/11.5.0/Firebase.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/0/3/5/Firebase/11.4.0/Firebase.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/0/3/5/Firebase/11.4.0/Firebase.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/0/3/5/Firebase/11.4.1/Firebase.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/0/3/5/Firebase/11.4.2/Firebase.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/0/3/5/Firebase/11.4.0/Firebase.podspec.json exists! Returning local because checking is only performed in repo update
    [!] CocoaPods could not find compatible versions for pod "Firebase/Analytics":
      In Podfile:
        firebase_analytics (from `Flutter/ephemeral/.symlinks/plugins/firebase_analytics/macos`) was resolved to 11.3.5, which depends on
          Firebase/Analytics (= 11.4.0)

    Specs satisfying the `Firebase/Analytics (= 11.4.0)` dependency were found, but they required a higher minimum deployment target.

    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:317:in `raise_error_unless_state'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:299:in `block in unwind_for_conflict'
    <internal:kernel>:90:in `tap'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:297:in `unwind_for_conflict'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:682:in `attempt_to_activate'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:254:in `process_topmost_state'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:182:in `resolve'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/molinillo-0.8.0/lib/molinillo/resolver.rb:43:in `resolve'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/cocoapods-1.16.2/lib/cocoapods/resolver.rb:94:in `resolve'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/cocoapods-1.16.2/lib/cocoapods/installer/analyzer.rb:1082:in `block in resolve_dependencies'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/cocoapods-1.16.2/lib/cocoapods/user_interface.rb:64:in `section'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/cocoapods-1.16.2/lib/cocoapods/installer/analyzer.rb:1080:in `resolve_dependencies'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/cocoapods-1.16.2/lib/cocoapods/installer/analyzer.rb:125:in `analyze'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/cocoapods-1.16.2/lib/cocoapods/installer.rb:422:in `analyze'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/cocoapods-1.16.2/lib/cocoapods/installer.rb:244:in `block in resolve_dependencies'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/cocoapods-1.16.2/lib/cocoapods/user_interface.rb:64:in `section'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/cocoapods-1.16.2/lib/cocoapods/installer.rb:243:in `resolve_dependencies'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/cocoapods-1.16.2/lib/cocoapods/installer.rb:162:in `install!'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/cocoapods-1.16.2/lib/cocoapods/command/install.rb:52:in `run'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/claide-1.1.0/lib/claide/command.rb:334:in `run'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/cocoapods-1.16.2/lib/cocoapods/command.rb:52:in `run'
    /Users/kaye375751/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/cocoapods-1.16.2/bin/pod:55:in `<top (required)>'
    /Users/kaye375751/.rbenv/versions/3.2.2/bin/pod:25:in `load'
    /Users/kaye375751/.rbenv/versions/3.2.2/bin/pod:25:in `<main>'

Error: The pod "Firebase/Analytics" required by the plugin "firebase_analytics" requires a higher minimum macOS deployment version than the plugin's reported minimum version.
To build, remove the plugin "firebase_analytics", or contact the plugin's developers for assistance.
Error: Error running pod install
```

### Flutter dependencies

<details>
<summary>Expand <code>Flutter dependencies</code> snippet</summary>
<br>

```yaml
Dart SDK 3.5.4
Flutter SDK 3.24.5
fba_demo 1.0.0+1

dependencies:
- cupertino_icons 1.0.8
- firebase_analytics 11.3.5 [firebase_analytics_platform_interface firebase_analytics_web firebase_core firebase_core_platform_interface flutter]
- firebase_core 3.8.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]

dev dependencies:
- flutter_lints 4.0.0 [lints]
- flutter_test 0.0.0 [flutter test_api matcher path fake_async clock stack_trace vector_math leak_tracker_flutter_testing async boolean_selector characters collection leak_tracker leak_tracker_testing material_color_utilities meta source_span stream_channel string_scanner term_glyph vm_service]

transitive dependencies:
- _flutterfire_internals 1.3.46 [collection firebase_core firebase_core_platform_interface flutter meta]
- async 2.11.0 [collection meta]
- boolean_selector 2.1.1 [source_span string_scanner]
- characters 1.3.0
- clock 1.1.1
- collection 1.18.0
- fake_async 1.3.1 [clock collection]
- firebase_analytics_platform_interface 4.2.7 [_flutterfire_internals firebase_core flutter meta plugin_platform_interface]
- firebase_analytics_web 0.5.10+4 [_flutterfire_internals firebase_analytics_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins]
- firebase_core_platform_interface 5.3.0 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 2.18.1 [firebase_core_platform_interface flutter flutter_web_plugins meta web]
- flutter_web_plugins 0.0.0 [flutter characters collection material_color_utilities meta vector_math]
- leak_tracker 10.0.5 [clock collection meta path vm_service]
- leak_tracker_flutter_testing 3.0.5 [flutter leak_tracker leak_tracker_testing matcher meta]
- leak_tracker_testing 3.0.1 [leak_tracker matcher meta]
- lints 4.0.0
- matcher 0.12.16+1 [async meta stack_trace term_glyph test_api]
- material_color_utilities 0.11.1 [collection]
- meta 1.15.0
- path 1.9.0
- plugin_platform_interface 2.1.8 [meta]
- sky_engine 0.0.99
- source_span 1.10.0 [collection path term_glyph]
- stack_trace 1.11.1 [path]
- stream_channel 2.1.2 [async]
- string_scanner 1.2.0 [source_span]
- term_glyph 1.2.1
- test_api 0.7.2 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph]
- vector_math 2.1.4
- vm_service 14.2.5
- web 1.1.0
```

</details>

**Comment by bubnov** [comment]
Unfortunately, the issue is still here.

![CleanShot 2024-11-25 at 16 55 10@2x](https://github.com/user-attachments/assets/2495c171-bfaf-436e-8216-6091cffce7fa)

```bash
user@user macos % pod install --repo-update
Updating local specs repositories
Analyzing dependencies
firebase_analytics: Using Firebase SDK version '11.4.0' defined in 'firebase_core'
firebase_core: Using Firebase SDK version '11.4.0' defined in 'firebase_core'
Run script to upload symbols already exists.
firebase_crashlytics: Using Firebase SDK version '11.4.0' defined in 'firebase_core'
firebase_messaging: Using Firebase SDK version '11.4.0' defined in 'firebase_core'
[!] CocoaPods could not find compatible versions for pod "Firebase/Analytics":
  In Podfile:
    firebase_analytics (from `Flutter/ephemeral/.symlinks/plugins/firebase_analytics/macos`) was resolved to 11.3.5, which depends on
      Firebase/Analytics (= 11.4.0)

Specs satisfying the `Firebase/Analytics (= 11.4.0)` dependency were found, but they required a higher minimum deployment target.
```

**Comment by SelaseKay** [comment]
In your macOS directory, delete the `Pods` folder and `Podfile.lock` file. Then, add `platform :osx, '10.15'` at the top of your `Podfile` and try running the app again.

**Comment by bubnov** [comment]
This helped: `platform :osx, '10.15'`. 
Its works now. Thanks!

**Comment by charneykaye** [comment]
Confirmed, resolved by setting `platform :osx, '10.15'` in my macos Podfile
